src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -139,24 +140,34 @@ public virtual Transmission Dequeue()
                         return transmission;
                     }
                 }
-                catch (UnauthorizedAccessException)
+                catch (UnauthorizedAccessException uae)
                 {
                     if (file == null)
                     {
-                        return null; // because the process does not have permission to access the folder
+                        TelemetryChannelEventSource.Log.TransmissionStorageDequeueUnauthorizedAccessException(this.folder.Name, uae.ToString());
```

Comment:
why do you output the folder name, not the file name. Didn't we get exception for the specific file access?